From 91c689e706c0bcc2a9dc3397a5c1da2e95201aba Mon Sep 17 00:00:00 2001
From: Esme Povirk <esme@codeweavers.com>
Date: Thu, 19 Feb 2026 20:18:26 +0000
Subject: [PATCH] windowscodecs: Fix size check in gif copy_interlaced_pixels.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=59428
(cherry picked from commit 45190d46646bf6716f18ba5eb8bbc9bfe2102d7a)
---
 dlls/windowscodecs/gifformat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/windowscodecs/gifformat.c b/dlls/windowscodecs/gifformat.c
index 290ae6bf899..287a49f83e9 100644
--- a/dlls/windowscodecs/gifformat.c
+++ b/dlls/windowscodecs/gifformat.c
@@ -607,7 +607,7 @@ static HRESULT copy_interlaced_pixels(const BYTE *srcbuffer,
     if (dststride < rc->Width)
         return E_INVALIDARG;
 
-    if ((dststride * rc->Height) > dstbuffersize)
+    if ((dststride * (rc->Height - 1)) + rc->Width > dstbuffersize)
         return E_INVALIDARG;
 
     row_offset = rc->X;
